sklearn/preprocessing/tests/test_data.py

```diff
@@ -2076,7 +2078,7 @@ def test_power_transformer_strictly_positive_exception():
                          pt.fit, X_with_negatives)
 
     assert_raise_message(ValueError, not_positive_message,
-                         power_transform, X_with_negatives)
+                         power_transform, X_with_negatives, 'box-cox')
```

Comment:
why is this needed? The default value shouldn't change, right? Or do we want to start a cycle to change the default to yeo-johnson?

Reply:
I find it clearer and explicit?

I don't know if we'll change the default but it should still be fine as PowerTransform hasn't been released yet AFAIK

Reply:
good point. We should discuss before the release. I think yeo-johnson would make more sense.

Reply:
The fact that Yeo-Johnson accepts negative values while Box-Cox does not makes me feel like we should use it by default. From a usability point of view, it's nicer to our users.

Reply:
I have the same feeling. Plus, it is designed to be a generalization of Box-Cox, even though that's not strictly the case.

Reply:
+1

Reply:
shall I change the default then?

Reply:
think so.

Reply:
Done